service/src/main/resources/db/changesets/20230606_postgres_rename_columns_and_constraints.yaml

```diff
@@ -0,0 +1,59 @@
+databaseChangeLog:
+  - changeSet:
+      id: postgres_rename_columns_and_constraints
+      author: chenchalsubraveti
+      dbms: postgresql
+      changes:
+        # for study_properties table
+        - dropUniqueConstraint:
+            constraintName: pk_sp
+            tableName: study_property
+        - renameColumn:
+            newColumnName: property_key
+            oldColumnName: key
+            tableName: study_property
+        - renameColumn:
+            newColumnName: property_value
+            oldColumnName: value
+            tableName: study_property
+        - addUniqueConstraint:
+            constraintName: pk_sp
+            tableName: study_property
+            columnNames: study_id, property_key
+
+        # for criteria_tag table
+        -  dropForeignKeyConstraint:
+             baseTableName:  criteria_tag
+             constraintName:  ck_crit_cr
+        -  addForeignKeyConstraint:
+             baseColumnNames:  cohort_revision_id
+             baseTableName:  criteria_tag
+             constraintName:  ck_crit_cr_2
+             onDelete:  CASCADE
+             referencedColumnNames:  id
+             referencedTableName:  cohort_revision
+        -  dropForeignKeyConstraint:
+             baseTableName:  criteria_tag
+             constraintName:  fk_crit_cs
+        -  addForeignKeyConstraint:
+             baseColumnNames:  concept_set_id
+             baseTableName:  criteria_tag
+             constraintName:  fk_crit_cs_2
+             onDelete:  CASCADE
+             referencedColumnNames:  id
+             referencedTableName:  concept_set
```

Comment:
I understand that you needed to rename the unique constraints because they included the renamed key->criteria_key column. I didn't expect that you'd also have to rename these FK constraints, since none of the relevant column names changed. Any idea why? Maybe the FK constraints somehow indirectly reference the old unique constraint? Just curious.

Reply:
Did you end up needing these after all?

Reply:
I did not try to rename and check. Will do and resolve, it may not be needed as it deals with only postgres - but have not checked

Reply:
I did not try to rename and check. Will do and resolve, it may not be needed as it deals with only postgres - but have not checked